### DISCUSSION TITLE
Dynamic operations, queries, mutations, subscriptions with Strawberry Shake

### DISCUSSION THREAD

**Discussion by frabe1579** [ChilliCream/hotchocolate#4472] [Q&A]
With Strawberry Shake, is there a way to invoke dynamic operations?
I have an app where the end-user can choose which columns to show in a table, can I pass to a query the list of visible fields, so that only the visible data is fetched from the server?
The returned data can be a list of elements of type Dictionary<string, JsonElement> or a JsonDocument, instead of a POCO.
Actualy the query string and the POCOs are built at compile time, and I see no way to change them.
Maybe there is one?
Thanks.

**Comment by michaelstaib** [comment]
In GraphQL skip and include are used for this purpose.

**Reply by frabe1579** [reply]
That's not feasible for my context, because the end user can potentially have to choose one or two additional fields from a list of tens or potentially hundreds. Skip and include directives seems to be not a solution. Also at the server side there would be additional fields that are unknown at compile time of the client.
No other way to generate a query/subscription at runtime? Maybe via reflection or expressions?
I've tried to go deep in the code but the caching layers confused me. Any suggestion on where to look at?
At the moment I'm using the simplets graphql client, but Strawberry Shake seems to be a good alternative for the vast majority of the cases, excluding dynamic queries.
Thank you.

**Comment by jimitndiaye** [comment]
I have a similar use case where clients can compose their own queries, including filters, sorting and projections. Has support for dynamic queries changed at all?

**Reply by michaelstaib** [reply]
Nope, it is nothing that we are implementing. In this case I would just craft the query and use http.
